coap-gateway/service/observation/deviceObserver.go

```diff
@@ -51,11 +51,12 @@ type DeviceObserver struct {
 }
 
 type DeviceObserverConfig struct {
-	ObservationType            ObservationType
-	TwinEnabled                bool
-	TwinEnabledSet             bool
-	Logger                     log.Logger
-	RequireBatchObserveEnabled bool
+	ObservationType              ObservationType
+	TwinEnabled                  bool
+	TwinEnabledSet               bool
+	Logger                       log.Logger
+	RequireBatchObserveEnabled   bool
+	UpdateDeviceMetadataResponse *commands.UpdateDeviceMetadataResponse
```

Comment:
Seems a bit weird to have a response in a configuration struct. What's the use case? I don't see it used anywhere.

Reply:
removed